refactor(ecmascript)!: remove `PropertyReadSideEffects::OnlyMemberPropertyAccess`

sapphi-red opened the pull request:
In #10129, I introduced `PropertyReadSideEffects`. In that, I added `PropertyReadSideEffects::OnlyMemberPropertyAccess`. This was because I thought this was the default behavior of Rollup at that time.

But after re-reading the Rollup's code and checking the behavior in REPL while reviewing https://github.com/rolldown/rolldown/pull/5945, I noticed that I was wrong. Since this was added solely for Rolldown to add support for `treeshake.propertyReadSideEffects` and it's not useful for that usecase, I removed the value.

cc @IWANABETHATGUY

sapphi-red commented:
* **#13348** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13348?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/13348?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=13348).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

IWANABETHATGUY commented on crates/oxc_ecmascript/src/side_effects/context.rs:
```diff
@@ -6,9 +6,6 @@ use crate::GlobalContext;
-    OnlyMemberPropertyAccess,
     /// Treat all property read accesses as possible side effects.
     #[default]
     All,
```
Should we also rename `All` to `Always` to align with rolldown/rollup?

IWANABETHATGUY commented:
https://deepwiki.com/search/what-is-the-difference-between_7a582090-34a7-4c97-a981-9bd6deeee882, here is the example deepwiki explained what is the difference between `true` and `always`

sapphi-red committed:
refactor(ecmascript)!: remove `PropertyReadSideEffects::OnlyMemberPropertyAccess` (#13348)

In #10129, I introduced `PropertyReadSideEffects`. In that, I added `PropertyReadSideEffects::OnlyMemberPropertyAccess`. This was because I thought this was the default behavior of Rollup at that time.

But after re-reading the Rollup's code and checking the behavior in REPL while reviewing https://github.com/rolldown/rolldown/pull/5945, I noticed that I was wrong. Since this was added solely for Rolldown to add support for `treeshake.propertyReadSideEffects` and it's not useful for that usecase, I removed the value.

cc @IWANABETHATGUY
